### PR TITLE
api: drop sgls from API

### PR DESCRIPTION
**Pull request by safl**
The level in which xNVMe encapsulates payloads are contigous buffers and
iovecs. As these are "layouts" commonly used for storage in user-facing
libraries, that is, where the user is another library or an application
running in user space.

The SGL representation and helper functions to create sgl-lists are
currently only usable with the SPDK/NVMe driver backend. This breaks the
goal of a unified API, ideally xNVMe should be providing contig. buffers
and iovecs and leave the PRP and SGL construction either to the backend,
or to the driver. Thus, this change to remove them.

Should it be requested, then SGLs could be re-introduced, along with
proper testing of their functionality and examples of their use.
Ideally, re-introduced in a manner compatible with other user-space NVMe
drivers, as the current implementation is specific to the SPDK/NVMe
driver.

Signed-off-by: Simon A. F. Lund <simon.lund@samsung.com>

### CODE REVIEW

**Review by karlowich — APPROVED** [review]
Looks good!

File: python/xnvme-cy-header/aux/generate_cython_header.py [hunk 1]
```diff
@@ -55,7 +55,6 @@ def generate_pxd(workspace_root):
         "libxnvme_znd.h",
         "libxnvme_pp.h",
         "libxnvme_be.h",
-        "libxnvme_sgl.h",
```
Comment (ankit-sam) [inline python/xnvme-cy-header/aux/generate_cython_header.py:58]: There are traces of xnvme_sgl here (Line 9, 10). they can be removed

**Review by ankit-sam — COMMENTED** [review]
Changes look good.

Small nit - few spelling mistakes in commit message

- librariy
- contigous

**Comment by safl** [comment]
Thanks @ankit-sam , fixed!

File: python/xnvme-cy-header/aux/generate_cython_header.py (hunk 1 above)
Comment (safl) [reply python/xnvme-cy-header/aux/generate_cython_header.py:58]: Overlooked this, fixed now. Also, found another mention of the SGLs in the cython header. Removed that as well.

Comment (Baekalfen) [reply python/xnvme-cy-header/aux/generate_cython_header.py:58]: Nice, to get rid of some sgl-specific fixes as well.

**Review by ankit-sam — APPROVED** [review]

**Review by karlowich — APPROVED** [review]

**Pull request merged**